spec/controllers/admin/accounts_controller_spec.rb

```diff
@@ -215,13 +215,8 @@
       it 'succeeds in rejecting account and logs action' do
         expect(subject).to redirect_to admin_accounts_path(status: 'pending')
 
-        expect(latest_admin_action_log)
-          .to be_present
-          .and have_attributes(
-            action: eq(:reject),
-            account_id: eq(current_user.account_id),
-            target_id: eq(account.user.id)
-          )
+        expect { account.reload }
+          .to raise_error(ActiveRecord::RecordNotFound)
```

Comment:
Seems like the test was missing a test on the state of the account, but is now missing the part about the log.

Reply:
Ah, thanks for asking about this, I had meant to leave inline notes.

For the two removed specs which previously checked for the action log ... these are both happening in "reject" actions, where the user record in question is sent to the `DeleteAccountService` by the controller action (with `reserve_email: false` in both cases, which means the user record will be deleted) before `log_action` is called.

With this change, the log record is no longer created ... which seems like a bugfix to me? ... hence I removed that from the spec.

If that's NOT a bug and we actively want to create a log with a reference to a record which will have been deleted (or I'm misunderstanding something here) ... let me know and I can adjust this to handle possible missing records.

I could also probably pull out 2-3 precursor PRs from this, and get this one to be JUST the changes to the concern and its spec -- everything else could be justified on its own.

Reply:
Pushed a change which restores the behavior of creating action logs for deleted records.